src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -135,6 +143,13 @@ public StreamingResponseBody streamEvents(
             SubscriptionStreamer streamer = null;
             final SubscriptionOutputImpl output = new SubscriptionOutputImpl(response, outputStream);
             try {
+                if  (subscriptionDbRepository.getSubscription(subscriptionId).getEventTypes().stream()
+                        .map(etName -> floodService.isConsumptionBlocked(etName)).findFirst().orElse(false)) {
+                    response.setStatus(HttpStatus.TOO_MANY_REQUESTS.value());
```

Comment:
Generally speaking, when I thought about situations in which the exceptions are thrown are critical for the whole nakadi component. Maybe it will be a good approach just to swallow all exceptions within the service level.